File: web-admin/src/api/maintenance-team.ts
import { api } from './http';

export interface MaintenanceTeamMember {
    id: string;
    team_id: string;
    user_id?: string;
    employee_id?: string;
    member_name: string;
    role_in_team: string;
    is_active: boolean;
    created_at: string;
}

export interface MaintenanceTeam {
    id: string;
    team_code: String;
    team_name: string;
    company_id?: string;
    company_name?: string;
    manager_id?: string;
    manager_name?: string;
    status: 'ACTIVE' | 'INACTIVE';
    description?: string;
    created_at: string;
    updated_at: string;
    members: MaintenanceTeamMember[];
    total_members: number;
}

export interface CreateTeamMemberPayload {
    user_id?: string;
    employee_id?: string;
    member_name: string;
    role_in_team: string;
}

export interface CreateMaintenanceTeamPayload {
    team_code: string;
    team_name: string;
    company_id?: string;
    manager_id?: string;
    manager_name?: string;
    description?: string;
    members: CreateTeamMemberPayload[];
}

export interface UpdateMaintenanceTeamPayload {
    team_code?: string;
    team_name?: string;
    company_id?: string;
    manager_id?: string;
    manager_name?: string;
    status?: 'ACTIVE' | 'INACTIVE';
    description?: string;
    members?: CreateTeamMemberPayload[];
}

export interface MaintenanceTeamQueryParams {
    search?: string;
    status?: string;
    company_id?: string;
    page?: number;
    per_page?: number;
}

export const maintenanceTeamApi = {
    list: async (params?: MaintenanceTeamQueryParams) => {
        const response = await api.get('/maintenance-teams', { params });
        return response.data;
    },

    get: async (id: string): Promise<MaintenanceTeam> => {
        const response = await api.get(`/maintenance-teams/${id}`);
        return response.data;
    },

    create: async (data: CreateMaintenanceTeamPayload): Promise<MaintenanceTeam> => {
        const response = await api.post('/maintenance-teams', data);
        return response.data;
    },

    update: async (id: string, data: UpdateMaintenanceTeamPayload): Promise<MaintenanceTeam> => {
        const response = await api.put(`/maintenance-teams/${id}`, data);
        return response.data;
    },

    delete: async (id: string): Promise<void> => {
        await api.delete(`/maintenance-teams/${id}`);
    },

    addMember: async (teamId: string, member: CreateTeamMemberPayload): Promise<MaintenanceTeamMember> => {
        const response = await api.post(`/maintenance-teams/${teamId}/members`, member);
        return response.data;
    },

    removeMember: async (teamId: string, memberId: string): Promise<void> => {
        await api.delete(`/maintenance-teams/${teamId}/members/${memberId}`);
    },
};

File: crates/core/src/domain/entities/maintenance_team.rs
//! Maintenance Team Domain Entity

use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use utoipa::ToSchema;
use uuid::Uuid;

/// Maintenance Team Entity
#[derive(Debug, Clone, Serialize, Deserialize, FromRow, ToSchema)]
pub struct MaintenanceTeam {
    pub id: Uuid,
    pub team_code: String,
    pub team_name: String,
    #[sqlx(default)]
    pub company_id: Option<Uuid>,
    #[sqlx(default)]
    pub manager_id: Option<Uuid>,
    #[sqlx(default)]
    pub manager_name: Option<String>,
    pub status: String,
    #[sqlx(default)]
    pub description: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

/// Maintenance Team Member Entity
#[derive(Debug, Clone, Serialize, Deserialize, FromRow, ToSchema)]
pub struct MaintenanceTeamMember {
    pub id: Uuid,
    pub team_id: Uuid,
    #[sqlx(default)]
    pub user_id: Option<Uuid>,
    #[sqlx(default)]
    pub employee_id: Option<Uuid>,
    pub member_name: String,
    pub role_in_team: String,
    pub is_active: bool,
    pub created_at: DateTime<Utc>,
}

/// Maintenance Team with Detail (including members list)
#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
pub struct MaintenanceTeamDetail {
    #[serde(flatten)]
    pub team: MaintenanceTeam,
    pub company_name: Option<String>,
    pub members: Vec<MaintenanceTeamMember>,
    pub total_members: usize,
}
